<launch>
	<include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="paused" value="true"/>
      <arg name="use_sim_time" value="true"/>
      <arg name="gui" value="true"/>
      <arg name="headless" value="false"/> 		<!-- Inert - see gazebo_ros_pkgs issue #491 -->
      <arg name="recording" value="false"/>
      <arg name="debug" value="false"/>
    </include>

    <!-- Spawn a robot into Gazebo -->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find final_project)/robots/B_SCARA.urdf.xacro"/>
    <node name="spawn" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model SCARA_T7 -unpause"/>

    <!-- Launch controller for robot joints -->
    <rosparam file="$(find final_project)/config/B_SCARA_config.yaml" command="load"/>

    <!-- load the controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/SCARA_T7" args=
      "joint_state_controller
       joint3_position_controller"/>

    <!-- convert joint states to TF transforms -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="rob_st_pub" respawn="false" output="screen">
    	<remap from="/joint_states" to="/SCARA_T7/joint_states" />
    </node> 

</launch>
